backend/src/main/kotlin/com/infowings/catalog/data/objekt/ObjectValidator.kt

```diff
@@ -172,27 +170,33 @@ class MainObjectValidator(
         }
 
         val aspectPropertyVertex = request.aspectPropertyId?.let {
-            aspectDao.findProperty(it)
-                    ?: throw AspectPropertyDoesNotExist(it)
+            aspectDao.findProperty(it) ?: throw AspectPropertyDoesNotExist(it)
         }
 
         val parentValueVertex = request.parentValueId?.let { objectService.findPropertyValueById(it) }
 
-        val btStr = if (request.parentValueId != null)
-            aspectPropertyVertex?.let { aspectDao.baseType(it) }
-        else objectPropertyVertex.let { objectDaoService.baseType(it) }
+        val valueAspectVertex = when (aspectPropertyVertex) {
+            null -> objectPropertyVertex.aspect ?: throw IllegalArgumentException("Object property has no reference to aspect")
+            else -> aspectPropertyVertex.associatedAspect
+        }
+
+        val btStr = valueAspectVertex.baseType ?: throw IllegalStateException("Associated aspect has no base type")
+        validateValueAssignableToBaseType(btStr, request.value)
```

Comment:
For patterns like "take some value from vertex or throw standard exception if it is null" it makes sense to define accessor with name like baseTypeStrict